core/model/modx/processors/browser/file/remove.class.php

```diff
@@ -33,7 +33,12 @@ public function process() {
         if (empty($file)) {
             return $this->modx->error->failure($this->modx->lexicon('file_err_ns'));
         }
-        $file = preg_replace('/[\.]{2,}/', '', $file);
+        $directory = preg_replace('/[\.]{2,}/', '', htmlspecialchars(pathinfo($file, PATHINFO_DIRNAME)));
+        if (!empty($directory)) {
+            $directory .= DIRECTORY_SEPARATOR;
+        }
+        $name = htmlspecialchars(end(explode( DIRECTORY_SEPARATOR, $file )));
```

Comment:
You could use pathinfo for both parts.

Reply:
I did so the first time.
Pathinfo with PATHINFO_BASENAME and PATHINFO_FILENAME  have trouble with filename with unciode symbols and more dots.
So after testing and catch more bugs I replace this on more simple - end->explode for filename.

e.g. http://codepad.org/73SaPqcF, first 4 characters in filename are loses.

Reply:
pathinfo is locale aware, so you have to set an UTF8 capable locale before:

```
$oldlocale = setlocale(LC_ALL, 0);
setlocale(LC_ALL,'C.UTF-8');
$pathinfo = pathinfo($file);
setlocale(LC_ALL,$oldlocale);
```

Reply:
See: http://codepad.org/nz0uth22

Reply:
You have to check, if 'C.UTF-8' is available.